release/tar/linux/opensearch-dashboards-tar-build.sh

```diff
@@ -0,0 +1,153 @@
+    echo "Arguments:"
+    echo -e "-f MANIFEST_FILE\t[Required] Path/to/manifest.yml file"
+    echo -e "-h help"
+}
+
+while getopts ":hf:" arg; do
+    case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        f)
+            MANIFEST_FILE=$OPTARG
+            ;;
+        :)
+            echo "Error: -${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$MANIFEST_FILE" ]; then
+    echo "error: You must specify the manifest file"
+    usage
+    exit 1
+fi
+
+if [ ! -f "$MANIFEST_FILE" ]; then
+    echo "error: The given file doesnot exist. Please check the specified path"
+    usage
+    exit 1
+fi
+
+SCHEMA_VERSION=`yq eval '.manifest.schema-version' $MANIFEST_FILE`
+PLATFORM=`yq eval '.build.platform' $MANIFEST_FILE`
+ARCHITECTURE=`yq eval '.build.architecture' $MANIFEST_FILE`
+VERSION=`yq eval '.build.version' $MANIFEST_FILE`
+UNDERLYING_ARCH=`uname -p`
+
+if [ -z "$VERSION" ]; then
+    echo "error: Please specify the version in the manifest file"
+    usage
+    exit 1
+fi
+
+if [ "$SCHEMA_VERSION" != "1.0" ]; then
+    echo "error: This script only supports manifest schema version 1.0. Please use the manifest file with right schema version"
+    usage
+    exit 1
+fi
+
+if [ "$PLATFORM" != "linux" ]; then
+    echo "error: $PLATFORM platform is not supported yet! Only supported platform is linux"
+    usage
+    exit 1
+fi
+
+if [ "$ARCHITECTURE" != "x64" ] && [ "$ARCHITECTURE" != "arm64" ]
+then
+    echo "error: $ARCHITECTURE architecture is not supported yet! Only supported architectures are x64 and arm64"
+    usage
+    exit 1
+fi
+
+if [[ ( "$ARCHITECTURE" == "arm64" && ( "$UNDERLYING_ARCH" != "aarch64" && "$UNDERLYING_ARCH" != "arm64" )) ||  ( "$ARCHITECTURE" == "x64" && ( "$UNDERLYING_ARCH" != "x86_64" && "$UNDERLYING_ARCH" != "amd64" )) ]]
+then
+    echo "error: The underlying architecture($UNDERLYING_ARCH) doesnot match the product architecture($ARCHITECTURE) that you are trying to build"
+    echo "To build $ARCHITECTURE you need to be on an $ARCHITECTURE architecture environment"
+    exit 1
+fi
+
+REPO_ROOT=`git rev-parse --show-toplevel`
+ROOT=`dirname $(realpath $0)`; cd $ROOT
+DIR_NAME="opensearch-dashboards-$VERSION"
+WORKING_DIR="$ROOT/$DIR_NAME"
+TARGET_DIR="$ROOT/target"
+PLUGINS_TEMP=`mktemp -d`
+OPENSEARCH_DASHBOARDS_CORE_URL=`yq eval '.products.opensearch-dashboards.opensearch-dashboards-min' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+OPENSEARCH_DASHBOARDS_PLUGINS_URLS=`yq eval '.products.opensearch-dashboards.plugins' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+
+TRAP_SIG_LIST="TERM INT EXIT CHLD"
+function delete_temp_folders() {
+rm -rf $WORKING_DIR $PLUGINS_TEMP $TARGET_DIR
+}
+trap delete_temp_folders $TRAP_SIG_LIST
+
+# # Download plugins to local 
+mkdir $WORKING_DIR
+mkdir -p $TARGET_DIR
```

Comment:
Why `mkdir` for one and `mkdir -p` for the other?

Reply:
Because WORKING_DIR is removed at the end of the script or by trap if the script fails. We want TARGET_DIR to exist even after script finishes. Usecase: User runs both dashboards and opensearch script. The script should not fail saying it already exists. (Same with what we did with /data folder) The reason for target folder is we have 2 artifacts 1 bundle and 1 sha512. It would easily get lost of not put in the folder. Dont want user to search through his folders for artifacts

Reply:
Do we want the script to fail if $WORKING_DIR already exists? It's OK if we do, but it's not clear to me that we want that behavior.

Reply:
Well in that case, wont the script remove the folder during trap? If the folder is already there and we say exit this part of the code will remove it right? Referring to this part of the code: https://github.com/opensearch-project/opensearch-build/pull/75/files#diff-341213f87e3420ebd2beca37f8e09c45f47c74c39cfc32b7b47678d290362ae0R116

Reply:
Would it be safe to add `-p` to $WORKING_DIR as well?

Reply:
Adding `-p` is safe. The way `mkdir` works is that it will fail with an error if the path already exists, unless you specify `-p`.

So the question is: Why are we failing if WORKING_DIR exists, but not failing if TARGET_DIR exists? If the answer is "we should fail on both", that's fine. If the answer is "we should not fail on either", that's also fine. If the answer is "WORKING_DIR is different to TARGET_DIR and here is why we fail for WORKING_DIR but not TARGET_DIR", that's also fine - just add a comment explaining why.

Reply:
Got it. I'll add `-p` to be safe.

Reply:
Done